File: src/main/java/org/cimsbioko/utilities/NetUtils.java
package org.cimsbioko.utilities;

import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import org.cimsbioko.App;

import static android.content.Context.CONNECTIVITY_SERVICE;

public class NetUtils {

    private static ConnectivityManager getConnMan() {
        return (ConnectivityManager) App.getApp().getSystemService(CONNECTIVITY_SERVICE);
    }

    public static boolean isConnected() {
        ConnectivityManager cm = getConnMan();
        if (cm == null) {
            return false;
        }
        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
        if (activeNetwork == null) {
            return false;
        }
        return activeNetwork.isConnected();
    }

    public static boolean isWiFiConnected() {
        ConnectivityManager cm = getConnMan();
        if (cm == null) {
            return false;
        }
        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
        if (activeNetwork == null) {
            return false;
        }
        return activeNetwork.isConnected() && activeNetwork.getType() == ConnectivityManager.TYPE_WIFI;
    }
}

File: src/main/kotlin/org/cimsbioko/utilities/NetUtils.kt
package org.cimsbioko.utilities

import android.content.Context
import android.net.ConnectivityManager
import org.cimsbioko.App

object NetUtils {

    private val connMan: ConnectivityManager
        get() = App.instance.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager

    val isConnected: Boolean
        get() {
            val cm = connMan
            val activeNetwork = cm.activeNetworkInfo ?: return false
            return activeNetwork.isConnected
        }

    val isWiFiConnected: Boolean
        get() {
            val cm = connMan
            val activeNetwork = cm.activeNetworkInfo ?: return false
            return activeNetwork.isConnected && activeNetwork.type == ConnectivityManager.TYPE_WIFI
        }

}